extensions/windows-event-log/wel/WindowsEventLog.cpp

```diff
@@ -43,10 +45,10 @@ void WindowsEventLogMetadataImpl::renderMetadata() {
   const auto contextGuard = gsl::finally([&context](){
     EvtClose(context);
   });
-  if (!EvtRender(context, event_ptr_, EvtRenderEventValues, dwBufferSize, nullptr, &dwBufferUsed, &dwPropertyCount)) {
+  if (!EvtRender(context, event_ptr_, EvtRenderEventValues, dwBufferSize, rendered_values.get(), &dwBufferUsed, &dwPropertyCount)) {
     if (ERROR_INSUFFICIENT_BUFFER == (status = GetLastError())) {
       dwBufferSize = dwBufferUsed;
-      rendered_values = std::unique_ptr<EVT_VARIANT, utils::FreeDeleter>((PEVT_VARIANT)(malloc(dwBufferSize)));
+      rendered_values.reset((PEVT_VARIANT)(malloc(dwBufferSize)));
```

Comment:
if you check all api calls, you can see that it first assigns the required buffer size (here `dwBufferUsed `) to the variable storing the actual buffer size (here `dwBufferSize `) and then calls `malloc` with the actual buffer size (modulo if it is in bytes or chars)

do you think it would be an improvement in readability if we allocate based on the required buffer size and then assign it to the actual buffer size?

Reply:
Sorry, I missed that one. It's fine this way.